pkg/policies/binary/binary.go

```diff
@@ -0,0 +1,212 @@
+	// OptConfig is the standard org-level opt in/out config, RepoOverride applies to all
+	// config.
+	OptConfig config.OrgOptConfig `yaml:"optConfig"`
+
+	// Action defines which action to take, default log, other: issue...
+	Action string `yaml:"action"`
+}
+
+// RepoConfig is the repo-level config for this policy.
+type RepoConfig struct {
+	// OptConfig is the standard repo-level opt in/out config.
+	OptConfig config.RepoOptConfig `yaml:"optConfig"`
+
+	// Action overrides the same setting in org-level, only if present.
+	Action *string `yaml:"action"`
+}
+
+type mergedConfig struct {
+	Action string
+}
+
+type details struct {
+	Messages []checker.CheckDetail
+}
+
+var configFetchConfig func(context.Context, *github.Client, string, string, string, interface{}) error
+
+func init() {
+	configFetchConfig = config.FetchConfig
+}
+
+// Binary is the Binary Artifacts policy object, implements policydef.Policy.
+type Binary bool
+
+// NewBinary returns a new Binary Artifacts policy.
+func NewBinary() policydef.Policy {
+	var b Binary
+	return b
+}
+
+// Name returns the name of this policy, implementing policydef.Policy.Name()
+func (b Binary) Name() string {
+	return polName
+}
+
+type logger struct {
+	Messages2 []checker.CheckDetail
+}
+
+func (l *logger) Info(desc string, args ...interface{}) {
+	cd := checker.CheckDetail{Type: checker.DetailInfo, Msg: fmt.Sprintf(desc, args...)}
+	l.Messages2 = append(l.Messages2, cd)
+}
+
+func (l *logger) Warn(desc string, args ...interface{}) {
+	cd := checker.CheckDetail{Type: checker.DetailWarn, Msg: fmt.Sprintf(desc, args...)}
+	l.Messages2 = append(l.Messages2, cd)
+}
+
+func (l *logger) Debug(desc string, args ...interface{}) {
+	cd := checker.CheckDetail{Type: checker.DetailDebug, Msg: fmt.Sprintf(desc, args...)}
+	l.Messages2 = append(l.Messages2, cd)
+}
+
+// Check performs the polcy check for this policy based on the
+// configuration stored in the org/repo, implementing policydef.Policy.Check()
+func (b Binary) Check(ctx context.Context, c *github.Client, owner,
+	repo string) (*policydef.Result, error) {
+	oc, rc := getConfig(ctx, c, owner, repo)
+	enabled := config.IsEnabled(oc.OptConfig, rc.OptConfig, repo)
+	log.Info().
+		Str("org", owner).
+		Str("repo", repo).
+		Str("area", polName).
+		Bool("enabled", enabled).
+		Msg("Check repo enabled")
+
+	repoClient := githubrepo.CreateGithubRepoClient(ctx, c)
+	if err := repoClient.InitRepo(owner, repo); err != nil {
+		return nil, err
+	}
+	l := logger{}
+	cr := &checker.CheckRequest{
+		Ctx:         ctx,
+		Client:      c,
+		RepoClient:  repoClient,
+		HTTPClient:  nil,
+		Owner:       owner,
+		Repo:        repo,
+		GraphClient: nil,
+		Dlogger:     &l,
+	}
+	// TODO, likely this should be a "scorecard" policy that runs multiple checks
+	// here, and uses config to enable/disable checks.
+	res := checks.BinaryArtifacts(cr)
+	if res.Error2 != nil {
+		return nil, res.Error2
+	}
+
+	var notify string
+	if res.Score < checker.MaxResultScore {
```

Comment:
Should the value we compare to (`MaxResultScore`) be part of the policy?

Reply:
Yes! I see that binary artifacts is just using max/min right now, so went with that for now:
https://github.com/ossf/scorecard/blob/main/checks/binary_artifact.go#L41-L45

That said, I think this policy should be expanded so that the operator can list in config which checks to run and which score is considered a "pass" for each check